### ISSUE TITLE
Plugins Failed To Load: Could not read VST plug-in metadata

### ISSUE THREAD

**Issue by sammdu** [robbert-vdh/yabridge#2]
Hi,

I was really excited to try out your program but after symlinking the .so files, Bitwig Studio cannot read any of them, claiming the error below:
```
Error reading metadata for file Synth1 VST.so:com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Could not read VST plug-in metadata
64 bit plugin host reported errors: Pluginhost returned non zero exit code 255

32 bit plugin host reported errors: Pluginhost returned non zero exit code 255
```

* I set Bitwig to scan within the directory of the plugins, i.e. where the new .so files were created.   
* My OS is Ubuntu 18.04.2.   
* My wine install is from the official WineHQ repository, version 5.0.   
* I use Bitwig Studio 1 (yes I haven't upgraded it lol, but it works fine with my plugins on Windows, as well as VST plugins that are Linux-native)

Please take a look, thanks!

**Comment by robbert-vdh** [comment]
Thanks for giving yabridge a shot! Yeah I found out shorty after releasing this that the prebuilt binaries are not going to work on Ubuntu 18.04 (it will work on anything newer though). Ubuntu 18.04 uses glibc 2.27, while this has been compiled for glibc 2.28. I'll try to get a build up specifically for Ubuntu 18.04 somewhere later today.

**Comment by sammdu** [comment]
Great! Thanks for looking into it!

**Comment by robbert-vdh** [comment]
It somehow took me a full day's worth of effort, but I've added a version of yabridge 1.0.0 compiled for Ubuntu 18.04 to the [releases page](https://github.com/robbert-vdh/yabridge/releases). This should in theory now be done automatically for any new releses. Let me know if it works for you!

**Comment by sammdu** [comment]
Hey thanks for fixing the binary package.

The previous error seems to have disappeared, but a new one is coming up. After I added the symlinks and started Bitwig Studio, it keeps trying to "index the plugin directory", and in the console it says "Could not read directory contents: /home/samm/.local/share/yabridge/libyabridge.so"

Do you have any idea what may have caused that?

**Comment by robbert-vdh** [comment]
Does the file `/home/samm/.local/share/yabridge/libyabridge.so` exist? Where did you extract the archive to? And if you run `ls -Al` in `~/.wine/drive_c/Program Files/Steinberg/VstPlugins` (assuming that's where the plugins you're trying to bridge are), are the .so files linked to the right files? For instance, this is what that looks like for me (I have them symlinked to the files in my build directory so the paths look a bit weird):

```shell
-rw-r--r--  1 robbert robbert  9775104  6 aug  2019  SerumFX_x64.dll
lrwxrwxrwx  1 robbert robbert       64  2 mei 18:31  SerumFX_x64.so -> /home/robbert/Documenten/projecten/yabridge/build/libyabridge.so
-rw-r--r--  1 robbert robbert  9855488 23 jan 03:17  Serum_x64.dll
lrwxrwxrwx  1 robbert robbert       64  2 mei 18:31  Serum_x64.so -> /home/robbert/Documenten/projecten/yabridge/build/libyabridge.so
drwxr-xr-x  2 robbert robbert     4096  2 mei 18:31  TBProAudio
-rw-r--r--  1 robbert robbert  8437376 18 okt  2018 'TDR VOS SlickEQ GE.dll'
lrwxrwxrwx  1 robbert robbert       64  2 mei 18:31 'TDR VOS SlickEQ GE.so' -> /home/robbert/Documenten/projecten/yabridge/build/libyabridge.so
```

**Comment by sammdu** [comment]
So I have a Windows partition, and I'm loading plugins from there:
```
/media/samm/OS/Program Files/VSTPlugIns/TAL$ ls -lhatotal 26M
drwxrwxrwx 1 root root 4.0K May  4 17:34 .
drwxrwxrwx 1 root root  40K May  4 17:34 ..
-rwxrwxrwx 2 root root 183K Jun 20  2011 TAL-Bitcrusher.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-Bitcrusher.so -> /home/samm/.local/share/yabridge/libyabridge.so
-rwxrwxrwx 2 root root 5.0M Mar 31  2017 TAL-Chorus-LX-64.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-Chorus-LX-64.so -> /home/samm/.local/share/yabridge/libyabridge.so
-rwxrwxrwx 2 root root 3.0M Jun 20  2018 TAL-Filter-2-64.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-Filter-2-64.so -> /home/samm/.local/share/yabridge/libyabridge.so
-rwxrwxrwx 2 root root 6.1M Mar 31  2017 TAL-NoiseMaker-64.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-NoiseMaker-64.so -> /home/samm/.local/share/yabridge/libyabridge.so
-rwxrwxrwx 2 root root 5.0M Mar 31  2017 TAL-Reverb-2-64.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-Reverb-2-64.so -> /home/samm/.local/share/yabridge/libyabridge.so
-rwxrwxrwx 1 root root 3.5M Feb  7 05:11 TAL-Reverb-4-64.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-Reverb-4-64.so -> /home/samm/.local/share/yabridge/libyabridge.so
-rwxrwxrwx 2 root root 3.3M Apr 10  2018 TAL-Vocoder-2-64.dll
lrwxrwxrwx 1 root root   47 May  4 17:34 TAL-Vocoder-2-64.so -> /home/samm/.local/share/yabridge/libyabridge.so
```

**Comment by robbert-vdh** [comment]
Now that I think about it, did you by any change add one of those .so files to the plugin locations in BItwig, rather than the directory they're in? "Could not read directory contents: /home/samm/.local/share/yabridge/libyabridge.so" sounds like it thinks one of those .so files in your VstPlugins directory is a directory. This is how I've set it up myself:

![](https://i.imgur.com/UB4s2MZ.png)

**Comment by sammdu** [comment]
* This is my plugin location setup:
![image](https://user-images.githubusercontent.com/10665890/81022872-c014ec00-8e3c-11ea-93c6-1e69675fa1dc.png)
* Yes I did extract the yabridge archive into `~/.local/share/yabridge`:
```
~/.local/share/yabridge$ lsa
total 1.8M
drwxr-xr-x  2 samm samm 4.0K May  4 17:34 .
drwx------ 46 samm samm 4.0K May  4 17:48 ..
-rwxr-xr-x  1 samm samm 705K May  4 17:02 libyabridge.so
-rwxr-xr-x  1 samm samm  347 May  4 17:34 link-plugins.sh
-rw-r--r--  1 samm samm  19K May  4 17:02 README.md
-rwxr-xr-x  1 samm samm  708 May  4 17:02 yabridge-host-32.exe
-rwxr-xr-x  1 samm samm 527K May  4 17:02 yabridge-host-32.exe.so
-rwxr-xr-x  1 samm samm  705 May  4 17:02 yabridge-host.exe
-rwxr-xr-x  1 samm samm 538K May  4 17:02 yabridge-host.exe.so

```

**Comment by robbert-vdh** [comment]
That looks like everything is set up as it should be! It could just be that Bitwig Studio 1 just does not support symlinks. I would just create copies of libyabridge.so instead of making links to it and see if that works. And finally, just to make sure since you said you're using Wine 5.0, you're not using Wien 5.7, right? That version has a regression in the way it handles winelib applications (like yabridge).

**Comment by sammdu** [comment]
Okay, interestingly, I found some more error messages. I let Bitwig run and scan and eventually ran into this message in the terminal:
```
Error reading metadata for file TAL-Bitcrusher.so:com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: com.bitwig.flt.library.metadata.reader.exception.CouldNotReadMetadataException: could not read metadata: Could not read VST plug-in metadata
64 bit plugin host reported errors: Pluginhost returned non zero exit code 143
Error messages:

09:00:00 [TAL-Bitcrusher-4DOTUZln] Initializing yabridge version 1.0.0
09:00:00 [TAL-Bitcrusher-4DOTUZln] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
09:00:00 [TAL-Bitcrusher-4DOTUZln] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Bitcrusher.dll'
09:00:00 [TAL-Bitcrusher-4DOTUZln] socket:     '/tmp/yabridge-TAL-Bitcrusher-4DOTUZln.sock'
09:00:00 [TAL-Bitcrusher-4DOTUZln] wineprefix: '<default>'
09:00:00 [TAL-Bitcrusher-4DOTUZln] 
09:00:00 [TAL-Bitcrusher-4DOTUZln] Enabled features:
09:00:00 [TAL-Bitcrusher-4DOTUZln] - bitbridge support
09:00:00 [TAL-Bitcrusher-4DOTUZln]

32 bit plugin host reported errors: Pluginhost returned non zero exit code 255

```
* I did the copy method this time
* I exported the yabridge folder to `$PATH`
* Still no VST plugins show up in Bitwig
* Look at line 3 of the yabridge message, i.e. the plugin line; the path starts with a double forward slash `//`

Does that help with anything?

**Comment by robbert-vdh** [comment]
At least it's actually loading! The double slash is something I'll have to fix in the next version, but on Linux paths starting with two slashes are equivalent to regular absolute paths. You are still sure you're not running Wine 5.7, right? You can verify if Wine is working correctly by running `/home/samm/.local/share/yabridge/yabridge-host.exe` in a terminal. If everything works correctly, you should see a bunch of Wine debug messages followed the the following line:

```
Usage: yabridge-host.exe <vst_plugin_dll> <unix_domain_socket>
```

If that is indeed the case, then you could try starting Bitwig with the following command line:

```
env YABRIDGE_DEBUG_FILE=/tmp/yabridge.log YABRIDGE_DEBUG_LEVEL=2 bitwig-studio
```

If you then try to scan plugin, yabridge will print a bunch of debug info to `/tmp/yabridge.log`. If you upload that somewhere (i.e. to hastebin) then we might be able to figure out what's going on.

I do think this to just be an issue with a wrong version of Wine though.

**Comment by sammdu** [comment]
This is what happened when running  `yabridge-host.exe` directly and via wine.
```
~/.local/share/yabridge$ wine --version
wine-5.0

~/.local/share/yabridge$ wine64 --version
wine-5.0

~/.local/share/yabridge$ /home/samm/.local/share/yabridge/yabridge-host.exe
Usage: yabridge-host.exe <vst_plugin_dll> <unix_domain_socket>

~/.local/share/yabridge$ wine /home/samm/.local/share/yabridge/yabridge-host.exe
002d:err:module:__wine_process_init failed to load L"Z:\\home\\samm\\.local\\share\\yabridge\\yabridge-host.exe", error c000012f

~/.local/share/yabridge$ wine64 /home/samm/.local/share/yabridge/yabridge-host.exe
0009:err:module:__wine_process_init failed to load L"Z:\\home\\samm\\.local\\share\\yabridge\\yabridge-host.exe", error c000012f
```

**Comment by robbert-vdh** [comment]
Then it does looks like everything's fine. Maybe Bitwig Studio 1 is just doing something weird? You can generate a debug log file at `/tmp/yabridge.log` by running Bitwig using:

```shell
env YABRIDGE_DEBUG_FILE=/tmp/yabridge.log YABRIDGE_DEBUG_LEVEL=2 bitwig-studio
```

If you could pass me that file then maybe I can figure out what's wrong. I don't have Bitwig Studio 1.3 installed at the moment so I can't really try it out for myself yet.

**Comment by sammdu** [comment]
I collected the yabridge log file, and there doesn't seem to be much to it:
```
16:00:38 [TAL-Bitcrusher-4EPYflnt] Initializing yabridge version 1.0.0
16:00:38 [TAL-Bitcrusher-4EPYflnt] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:00:38 [TAL-Bitcrusher-4EPYflnt] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Bitcrusher.dll'
16:00:38 [TAL-Bitcrusher-4EPYflnt] socket:     '/tmp/yabridge-TAL-Bitcrusher-4EPYflnt.sock'
16:00:38 [TAL-Bitcrusher-4EPYflnt] wineprefix: '<default>'
16:00:38 [TAL-Bitcrusher-4EPYflnt] 
16:00:38 [TAL-Bitcrusher-4EPYflnt] Enabled features:
16:00:38 [TAL-Bitcrusher-4EPYflnt] - bitbridge support
16:00:38 [TAL-Bitcrusher-4EPYflnt] 
16:05:38 [TAL-Chorus-LX-64-349GTYZe] Initializing yabridge version 1.0.0
16:05:38 [TAL-Chorus-LX-64-349GTYZe] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:05:38 [TAL-Chorus-LX-64-349GTYZe] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Chorus-LX-64.dll'
16:05:38 [TAL-Chorus-LX-64-349GTYZe] socket:     '/tmp/yabridge-TAL-Chorus-LX-64-349GTYZe.sock'
16:05:38 [TAL-Chorus-LX-64-349GTYZe] wineprefix: '<default>'
16:05:38 [TAL-Chorus-LX-64-349GTYZe] 
16:05:38 [TAL-Chorus-LX-64-349GTYZe] Enabled features:
16:05:38 [TAL-Chorus-LX-64-349GTYZe] - bitbridge support
16:05:38 [TAL-Chorus-LX-64-349GTYZe] 
16:10:38 [TAL-Filter-2-64-017JNUVe] Initializing yabridge version 1.0.0
16:10:38 [TAL-Filter-2-64-017JNUVe] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:10:38 [TAL-Filter-2-64-017JNUVe] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Filter-2-64.dll'
16:10:38 [TAL-Filter-2-64-017JNUVe] socket:     '/tmp/yabridge-TAL-Filter-2-64-017JNUVe.sock'
16:10:38 [TAL-Filter-2-64-017JNUVe] wineprefix: '<default>'
16:10:38 [TAL-Filter-2-64-017JNUVe] 
16:10:38 [TAL-Filter-2-64-017JNUVe] Enabled features:
16:10:38 [TAL-Filter-2-64-017JNUVe] - bitbridge support
16:10:38 [TAL-Filter-2-64-017JNUVe] 
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] Initializing yabridge version 1.0.0
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-NoiseMaker-64.dll'
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] socket:     '/tmp/yabridge-TAL-NoiseMaker-64-49DSYqrw.sock'
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] wineprefix: '<default>'
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] 
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] Enabled features:
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] - bitbridge support
16:15:38 [TAL-NoiseMaker-64-49DSYqrw] 
16:20:38 [TAL-Reverb-2-64-57Bdehqu] Initializing yabridge version 1.0.0
16:20:38 [TAL-Reverb-2-64-57Bdehqu] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:20:38 [TAL-Reverb-2-64-57Bdehqu] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Reverb-2-64.dll'
16:20:38 [TAL-Reverb-2-64-57Bdehqu] socket:     '/tmp/yabridge-TAL-Reverb-2-64-57Bdehqu.sock'
16:20:38 [TAL-Reverb-2-64-57Bdehqu] wineprefix: '<default>'
16:20:38 [TAL-Reverb-2-64-57Bdehqu] 
16:20:38 [TAL-Reverb-2-64-57Bdehqu] Enabled features:
16:20:38 [TAL-Reverb-2-64-57Bdehqu] - bitbridge support
16:20:38 [TAL-Reverb-2-64-57Bdehqu] 
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] Initializing yabridge version 1.0.0
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Reverb-4-64.dll'
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] socket:     '/tmp/yabridge-TAL-Reverb-4-64-7HJRXcjv.sock'
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] wineprefix: '<default>'
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] 
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] Enabled features:
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] - bitbridge support
16:25:38 [TAL-Reverb-4-64-7HJRXcjv] 
16:30:38 [TAL-Vocoder-2-64-18CJKenv] Initializing yabridge version 1.0.0
16:30:38 [TAL-Vocoder-2-64-18CJKenv] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
16:30:38 [TAL-Vocoder-2-64-18CJKenv] plugin:     '//media/samm/OS/Program Files/VSTPlugIns/TAL/TAL-Vocoder-2-64.dll'
16:30:38 [TAL-Vocoder-2-64-18CJKenv] socket:     '/tmp/yabridge-TAL-Vocoder-2-64-18CJKenv.sock'
16:30:38 [TAL-Vocoder-2-64-18CJKenv] wineprefix: '<default>'
16:30:38 [TAL-Vocoder-2-64-18CJKenv] 
16:30:38 [TAL-Vocoder-2-64-18CJKenv] Enabled features:
16:30:38 [TAL-Vocoder-2-64-18CJKenv] - bitbridge support
16:30:38 [TAL-Vocoder-2-64-18CJKenv] 
```
That's really weird. But there seems to me more information with the console log of Bitwig Studio, you can see it here: https://0bin.net/paste/CZImSyTRkS9no7FV#4HQdejgeP-fcAiJKXtVtcMLW1fE3H6/JLKNQX8xV48x

**Comment by siborg** [comment]
Hi,

I've had a similar issue yesterday, please close Bitwig, make sure no
instances of yabridge are left running and no wine stale processes are
showing when you run ps aux|grep wine . Once you confirmed that, please
check if you see any leftover sck files from yabridge. deleting them
resolved the issue for me.


Very Best Regards,

Szymon Kosecki

**Comment by robbert-vdh** [comment]
That looks like the actual plugin host process did not even start. I would again make 100% sure that you're running WIne 5.5 or 5.6, since Wine 5.7 still has problems. I'll try and see if I can built in even more ways to diagnose startup problems, because I'm not sure what's wrong here.

**Comment by robbert-vdh** [comment]
I just realized that all output from Wine is hidden before the sockets have been accepted (and thus before the plugin has finished initializing). My mistake. I'll upload a build that fixes this in a minute that you can try out.

**Comment by robbert-vdh** [comment]
This a build for Ubuntu 18.04 based on 750130e4a5f098730479db05f408ec040a0c7543: [yabridge-1.0.0-24-g750130e-ubuntu-18.04.tar.gz](https://github.com/robbert-vdh/yabridge/files/4583701/yabridge-1.0.0-24-g750130e-ubuntu-18.04.tar.gz)

It likely won't fix anything, but you should at least see errors from the Wine process now. And again, please make sure that you're running either Wine 5.5 or Wine 5.6 (earlier versions might also work fine, but Wine 5.7 won't work). And just run `wineserver -k` before trying to make sure nothing Wine related is still running.

**Comment by sammdu** [comment]
I downloaded the new build, deleted the old files, and remade copies of the `.so` files. Here is the `yabridge.log` file:
```
18:50:11 [TAL-Reverb-2-64-789QXatx] Initializing yabridge version 1.0.0-24-g750130e
18:50:11 [TAL-Reverb-2-64-789QXatx] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
18:50:11 [TAL-Reverb-2-64-789QXatx] plugin:     '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Reverb-2-64.dll'
18:50:11 [TAL-Reverb-2-64-789QXatx] socket:     '/tmp/yabridge-TAL-Reverb-2-64-789QXatx.sock'
18:50:11 [TAL-Reverb-2-64-789QXatx] wineprefix: '//home/samm/.wine'
18:50:11 [TAL-Reverb-2-64-789QXatx] 
18:50:11 [TAL-Reverb-2-64-789QXatx] Enabled features:
18:50:11 [TAL-Reverb-2-64-789QXatx] - bitbridge support
18:50:11 [TAL-Reverb-2-64-789QXatx] 
18:50:14 [TAL-Reverb-2-64-789QXatx] [Wine STDERR] Initializing yabridge host version 1.0.0-24-g750130e
18:50:14 [TAL-Reverb-2-64-789QXatx] [Wine STDERR] Error while initializing Wine VST host:
18:50:14 [TAL-Reverb-2-64-789QXatx] [Wine STDERR] Could not load a shared library at '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Reverb-2-64.dll'.
18:50:14 [TAL-Reverb-2-64-789QXatx] [Wine STDOUT] 
18:50:14 [TAL-Reverb-2-64-789QXatx] [Wine STDOUT] 
18:50:18 [TAL-Reverb-2-64-789QXatx] [Wine STDERR] 
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] Initializing yabridge version 1.0.0-24-g750130e
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] plugin:     '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Vocoder-2-64.dll'
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] socket:     '/tmp/yabridge-TAL-Vocoder-2-64-27Pfhmpx.sock'
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] wineprefix: '//home/samm/.wine'
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] 
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] Enabled features:
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] - bitbridge support
18:55:11 [TAL-Vocoder-2-64-27Pfhmpx] 
18:55:14 [TAL-Vocoder-2-64-27Pfhmpx] [Wine STDERR] Initializing yabridge host version 1.0.0-24-g750130e
18:55:14 [TAL-Vocoder-2-64-27Pfhmpx] [Wine STDERR] Error while initializing Wine VST host:
18:55:14 [TAL-Vocoder-2-64-27Pfhmpx] [Wine STDERR] Could not load a shared library at '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Vocoder-2-64.dll'.
18:55:14 [TAL-Vocoder-2-64-27Pfhmpx] [Wine STDOUT] 
18:55:14 [TAL-Vocoder-2-64-27Pfhmpx] [Wine STDOUT] 
18:55:21 [TAL-Vocoder-2-64-27Pfhmpx] [Wine STDERR] 
19:00:11 [TAL-Filter-2-64-ADGINScw] Initializing yabridge version 1.0.0-24-g750130e
19:00:11 [TAL-Filter-2-64-ADGINScw] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
19:00:11 [TAL-Filter-2-64-ADGINScw] plugin:     '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Filter-2-64.dll'
19:00:11 [TAL-Filter-2-64-ADGINScw] socket:     '/tmp/yabridge-TAL-Filter-2-64-ADGINScw.sock'
19:00:11 [TAL-Filter-2-64-ADGINScw] wineprefix: '//home/samm/.wine'
19:00:11 [TAL-Filter-2-64-ADGINScw] 
19:00:11 [TAL-Filter-2-64-ADGINScw] Enabled features:
19:00:11 [TAL-Filter-2-64-ADGINScw] - bitbridge support
19:00:11 [TAL-Filter-2-64-ADGINScw] 
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDERR] Initializing yabridge host version 1.0.0-24-g750130e
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDERR] Error while initializing Wine VST host:
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDERR] Could not load a shared library at '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Filter-2-64.dll'.
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDOUT] 
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDOUT] 
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDOUT] 
19:00:14 [TAL-Filter-2-64-ADGINScw] [Wine STDOUT] 
19:00:18 [TAL-Filter-2-64-ADGINScw] [Wine STDERR] 
19:05:11 [TAL-NoiseMaker-64-269EJOUi] Initializing yabridge version 1.0.0-24-g750130e
19:05:11 [TAL-NoiseMaker-64-269EJOUi] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
19:05:11 [TAL-NoiseMaker-64-269EJOUi] plugin:     '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-NoiseMaker-64.dll'
19:05:11 [TAL-NoiseMaker-64-269EJOUi] socket:     '/tmp/yabridge-TAL-NoiseMaker-64-269EJOUi.sock'
19:05:11 [TAL-NoiseMaker-64-269EJOUi] wineprefix: '//home/samm/.wine'
19:05:11 [TAL-NoiseMaker-64-269EJOUi] 
19:05:11 [TAL-NoiseMaker-64-269EJOUi] Enabled features:
19:05:11 [TAL-NoiseMaker-64-269EJOUi] - bitbridge support
19:05:11 [TAL-NoiseMaker-64-269EJOUi] 
19:05:13 [TAL-NoiseMaker-64-269EJOUi] [Wine STDERR] Initializing yabridge host version 1.0.0-24-g750130e
19:05:13 [TAL-NoiseMaker-64-269EJOUi] [Wine STDERR] Error while initializing Wine VST host:
19:05:13 [TAL-NoiseMaker-64-269EJOUi] [Wine STDERR] Could not load a shared library at '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-NoiseMaker-64.dll'.
19:05:13 [TAL-NoiseMaker-64-269EJOUi] [Wine STDOUT] 
19:05:17 [TAL-NoiseMaker-64-269EJOUi] [Wine STDERR] 
19:10:11 [TAL-Reverb-4-64-23CENPip] Initializing yabridge version 1.0.0-24-g750130e
19:10:11 [TAL-Reverb-4-64-23CENPip] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
19:10:11 [TAL-Reverb-4-64-23CENPip] plugin:     '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Reverb-4-64.dll'
19:10:11 [TAL-Reverb-4-64-23CENPip] socket:     '/tmp/yabridge-TAL-Reverb-4-64-23CENPip.sock'
19:10:11 [TAL-Reverb-4-64-23CENPip] wineprefix: '//home/samm/.wine'
19:10:11 [TAL-Reverb-4-64-23CENPip] 
19:10:11 [TAL-Reverb-4-64-23CENPip] Enabled features:
19:10:11 [TAL-Reverb-4-64-23CENPip] - bitbridge support
19:10:11 [TAL-Reverb-4-64-23CENPip] 
19:10:13 [TAL-Reverb-4-64-23CENPip] [Wine STDERR] Initializing yabridge host version 1.0.0-24-g750130e
19:10:13 [TAL-Reverb-4-64-23CENPip] [Wine STDERR] Error while initializing Wine VST host:
19:10:13 [TAL-Reverb-4-64-23CENPip] [Wine STDERR] Could not load a shared library at '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Reverb-4-64.dll'.
19:10:13 [TAL-Reverb-4-64-23CENPip] [Wine STDOUT] 
19:10:18 [TAL-Reverb-4-64-23CENPip] [Wine STDERR] 
19:15:11 [TAL-Chorus-LX-64-BPYZbest] Initializing yabridge version 1.0.0-24-g750130e
19:15:11 [TAL-Chorus-LX-64-BPYZbest] host:       '/home/samm/.local/share/yabridge/yabridge-host.exe'
19:15:11 [TAL-Chorus-LX-64-BPYZbest] plugin:     '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Chorus-LX-64.dll'
19:15:11 [TAL-Chorus-LX-64-BPYZbest] socket:     '/tmp/yabridge-TAL-Chorus-LX-64-BPYZbest.sock'
19:15:11 [TAL-Chorus-LX-64-BPYZbest] wineprefix: '//home/samm/.wine'
19:15:11 [TAL-Chorus-LX-64-BPYZbest] 
19:15:11 [TAL-Chorus-LX-64-BPYZbest] Enabled features:
19:15:11 [TAL-Chorus-LX-64-BPYZbest] - bitbridge support
19:15:11 [TAL-Chorus-LX-64-BPYZbest] 
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDERR] Initializing yabridge host version 1.0.0-24-g750130e
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDERR] Error while initializing Wine VST host:
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDERR] Could not load a shared library at '//home/samm/.wine/drive_c/Program Files/VSTPlugins/TAL/TAL-Chorus-LX-64.dll'.
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDOUT] 
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDOUT] 
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDOUT] 
19:15:13 [TAL-Chorus-LX-64-BPYZbest] [Wine STDOUT] 
19:15:17 [TAL-Chorus-LX-64-BPYZbest] [Wine STDERR] 
```

Interesting observation: I tried the airwave build from 2016 for comparison, and it seems like it works, but only if the plugins are under the `~/.wine/` folder, not in the Windows partition. That's also why you can see above that I copied my TAL plugins under `~/.wine/` but still no luck with yabridge.

**Comment by sammdu** [comment]
@siborg I tried your method but unfortunately it did not help. Thanks for suggesting!

**Comment by robbert-vdh** [comment]
The `Could not load a shared library at '*'`  error (I've changed the message new because the wording here is a bit confusing) is caused by Wine being unable to load the .DLL file. But I think I've figured it out. I'm not sure what is adding these double slashes, but even though that would be a completely valid path, Wine just won't load any files that start with a double forward slash.

Anyways, here is a build that strips out those double slashes. This should finally work! I'll figure out why this is happening and how to fix it properly tomorrow.

[yabridge-1.0.0-27-ge40f900-ubuntu-18.04.tar.gz](https://github.com/robbert-vdh/yabridge/files/4584314/yabridge-1.0.0-27-ge40f900-ubuntu-18.04.tar.gz)

**Comment by sammdu** [comment]
Yay! That did it! Not only does it work under the `~/.wine` folder, it also works under my Windows partition!
Thanks so much for bearing with me and helping fix this issue. You should publish this build to the releases page!

**Comment by sammdu** [comment]
Update: Bitwig Studio 1 does not support linking because it loads the first plugin and makes duplicates rather than loading the different plugin. Should be good info for the README file.

**Comment by robbert-vdh** [comment]
Great to hear it's actually working now! It was a bit of a shot in the dark, but now I do at least know the cause if this. For some reason, the `boost::dll:this_line()` function used to determine the location  of the copy to or symlink of `libyabridge.so` returns paths starting with a double forward slash on some Ubuntu based distros. This sounds like a bug in Boost or libdl, but it's something we can at least easily work around for the time being.

**Comment by robbert-vdh** [comment]
I've now added an actual fix for this issue. I'll release a new version with this fix and a few more useful features in the next few days!

**Comment by sammdu** [comment]
Great! I don't know if I should close the issue, so I'll leave it up to you.

**Comment by robbert-vdh** [comment]
The issue with paths starting with two slashes that was causing this should be fixed in yabridge 1.1.0, so I'm closing this issue. Feel free to reopen this issue if there are still any similar problems with paths.
